### PR TITLE
Use `HistoricalStdlibVersions` to populate `Pkg.Types.STDLIBS_BY_VERSION`

### PR DESCRIPTION
**Pull request by giordano**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -66,6 +66,8 @@ jobs:
       - uses: julia-actions/julia-buildpkg@latest
       - name: System info
         run: julia --project=. --color=yes -e "using BinaryBuilderBase; BinaryBuilderBase.versioninfo()"
+      - name: Install HistoricalStdlibVersions
+        run: julia --project=. --color=yes -e 'using Pkg; Pkg.add(; url="https://github.com/JuliaPackaging/HistoricalStdlibVersions.jl")'
```
Comment (giordano) [inline .github/workflows/ci.yml:70]: This is only while the package is being registered, no need to cut short the waiting time.

File: src/BinaryBuilderBase.jl [hunk 2]
```diff
@@ -219,6 +220,9 @@ function __init__()
     if get(ENV, "BINARYBUILDER_USE_CCACHE", "false") == "true"
         use_ccache[] = true
     end
+
+    # Populate `Pkg.STDLIBS_BY_VERSION`
+    append!(empty!(Pkg.STDLIBS_BY_VERSION), HistoricalStdlibVersions.STDLIBS_BY_VERSION)
```
Comment (IanButterworth) [inline src/BinaryBuilderBase.jl:225]:
```suggestion
    append!(empty!(Pkg.Types.STDLIBS_BY_VERSION), HistoricalStdlibVersions.STDLIBS_BY_VERSION)
```

**Review by DilumAluthge — COMMENTED** [review]
I think this should be done in the `BinaryBuilderBase.__init__()` function, right?

File: src/BinaryBuilderBase.jl (hunk 2 above)
Comment (IanButterworth) [reply src/BinaryBuilderBase.jl:225]: Sorry, was a typo in the readme. Was correct in the docstring

**Comment by giordano** [comment]
> I think this should be done in the `BinaryBuilderBase.__init__()` function, right?

Which is where it's done: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/759f3c04f08b7cb48db75a4fa656d5032cb6fee2/src/BinaryBuilderBase.jl#L170-L226 Or am I missing something?

File: src/BinaryBuilderBase.jl [hunk 3]
```diff
@@ -219,6 +220,9 @@ function __init__()
     if get(ENV, "BINARYBUILDER_USE_CCACHE", "false") == "true"
         use_ccache[] = true
     end
+
+    # Populate `Pkg.STDLIBS_BY_VERSION`
```
Comment (IanButterworth) [inline src/BinaryBuilderBase.jl:224]: comment nit
```suggestion
    # Populate `Pkg.Types.STDLIBS_BY_VERSION`
```

**Comment by DilumAluthge** [comment]
Ah, GitHub hid that part from me, so I couldn't tell that we were inside the `__init__` function.

**Review by DilumAluthge — APPROVED** [review]

**Comment by giordano** [comment]
Sounds like this block https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/1aae9a7b60d3bac5cf31114e3423be6b6617289f/test/compat.jl#L29-L45 is finally gone for good.

**Pull request merged**